app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -0,0 +1,148 @@
+package com.github.se.icebreakrr.model.profile
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.GeoPoint
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewModel() {
+
+  private val _profiles = MutableStateFlow<List<Profile>>(emptyList())
+  val profiles: StateFlow<List<Profile>> = _profiles
+
+  private val _selectedProfile = MutableStateFlow<Profile?>(null)
+  val selectedProfile: StateFlow<Profile?> = _selectedProfile
+
+  private val _loading = MutableStateFlow(false)
+  val loading: StateFlow<Boolean> = _loading
+
+  private val _error = MutableStateFlow<Exception?>(null)
+  val error: StateFlow<Exception?> = _error
+
+  init {
+    repository.init { // TODO getProfilesInRadius}
+    }
+  }
+
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ProfilesViewModel(ProfilesRepositoryFirestore(Firebase.firestore)) as T
+          }
+        }
+  }
+
+  /**
+   * Fetches profiles within a given radius around a center point with optional filters.
+   *
+   * @param center The center location as a GeoPoint.
+   * @param radiusInMeters The radius around the center in meters to search for profiles.
+   * @param gender Optional filter for gender.
+   * @param ageRange Optional filter for age range.
+   * @param tags Optional filter for specific tags.
+   */
+  fun getFilteredProfilesInRadius(
+      center: GeoPoint,
+      radiusInMeters: Double,
+      gender: Gender? = null,
+      ageRange: IntRange? = null,
+      tags: List<String>? = null
+  ) {
+    _loading.value = true
+    repository.getProfilesInRadius(
+        center = center,
+        radiusInMeters = radiusInMeters,
+        onSuccess = { profileList ->
+          val filteredProfiles =
+              profileList.filter { profile ->
+
+                // Filter by gender if specified
+                (gender == null || profile.gender == gender) &&
+
+                    // Filter by age range if specified
+                    (ageRange == null || profile.calculateAge() in ageRange) &&
+
+                    // Filter by tags if specified
+                    (tags == null || profile.tags.containsAll(tags))
```

Comment:
I think this should be containsAny rather than contains all.
If you set your preferences to football and pasta, it means that you can only see people who have set both football and pasta in their profile. 
The point would be more to see people who share one of your interests rather than all your interests.

Reply:
But this is temporary because Samuel is currently implementing, in his TagsViewModel, a filter such that when you are interested by football, which is in the category sports, you can still see the profiles of people interested by sports globally.

Reply:
I don't really agree with this remark. The classic behavior expected by the user with this type of filter is to correspond exactly to the request. For example if the user puts the tag "football" and "pizza" he expects to see only profiles which contain these two tags (in addition to others).